composer.json

```diff
@@ -42,7 +42,7 @@
         }
     ],
     "require": {
-        "php": "^7.2",
+        "php": "^7.2.9",
```

Comment:
I'm curious where this specific constraint comes from. Does it contain some bug fix that Infection relies on?

I maintain a package which requires `"php": "^7.2"` where I cannot upgrade Infection. While PHP 7.2 is still supported by its maintainers, I'm curious if I should start requiring `"php": "^7.2.9"` as well or not upgrade Infection until the PHP 7.2's EOL.

Reply:
How about that's the version Infection was tested to work with. If we'd advertise that we support everything down to PHP 7.2.0, that will mean that we got to be testing at all these versions. But we don't, so here we go. Otherwise put, this requirement is primarily a reflection of an actual deployment situation.

(Honestly I think it is about time to bump that requirement up to 7.3. Even Debian stable already has it.)

Reply:
The problem with you package is that you require that [platform must be PHP 7.2.0 exactly](https://github.com/morozov/kih/blob/master/composer.json#L13). Most of the time you don't need to declare a platform, Infection being one of the special cases where this is useful. E.g. if people install your package from composer, then you don't ever need to declare the platform, but if you're building a Phar that is supposed to work with a specific version of PHP, but you also intend to test that Phar under other versions of PHP like 7.3 or 7.4, then it is imperative to have this constrain because it makes sure every Phar you build even under PHP 7.3 is compatible with PHP 7.2.

Bottom line: remove your platform requirement and sleep well.

Reply:
@sanmai has it been tested with all versions between `7.2.10` and `7.2.21`? It should have been according to the argument above.

> (Honestly I think it is about time to bump that requirement up to 7.3. Even Debian stable already has it.)

It is but it's irrelevant to the question. By dropping PHP 7.2 you may rely on the features available in the newer release. By dropping PHP <= 7.2.8 you're not getting anything. The "actual deployment situation" does not reflect the actual dependency constraint `^7.2.9` which also includes `7.4` and `dev-master`.

Reply:
Well, no. I think I took the wrong turn with that my argument, because your problem is that you say you package is supposed to run under exactly PHP 7.2.0 by means of the platform directive. But is this true? What is the actual version that you use? I doubt it is exactly PHP 7.2.0.

Reply:
7.2.8 was released like, what, summer 2018, over a year ago. 7.2.9 was also released over a year ago. I don't see how this could be the actual problem here.

Reply:
> The problem with you package […]

Not being able to upgrade Infection is not my problem. I know _why_ I cannot upgrade it and was asking about the specific version constraint in Infection. This platform configuration exists in order to not let Composer install the dependencies which would break the `^7.2` requirement. E.g. if run `composer update` on PHP 7.3, it will install `ocramius/package-versions 1.5` which will require PHP 7.3 which I'd like to avoid.

> […] sleep well

As a package maintainer, it's not your concern. Please watch your tone.

Reply:
> If you feel like we must support installing Infection on PHP 7.2.0 exactly, you can certainly open a PR with that suggestion.

No. I was asking why exactly `^7.2.9`. It's up to the package maintainers to decide which versions to support.

Reply:
I recall this might be the version of the PHP 7.2 branch installed at that time. Makes sense?

Reply:
Installed where?

Reply:
It was on my development machine I used then. I'm sorry I do not exactly have time now for this or any interrogation, so let's not continue with it. You're totally free to open a PR with this change, and if there's a consensus, it might even get merged.

Reply:
I'm asking these questions to understand the reasoning behind this change while you're being too emotional. You're free to not answer them if it feels like an interrogation.

As I understand them so far, the reasons are:

1. PHP 7.2.8 and the older versions are too old.
2. The non-latest PHP versions have security issues, however, some of the non-latest PHP versions are still supported.
3. PHP 7.2.9 was installed on your development machine at some point in time.

It all makes sense. Thank you for the explanation.